keps/sig-storage/1710-selinux-relabeling/README.md

```diff
@@ -820,3 +815,29 @@ At the same time, `FSGroupChangePolicy: OnRootMismatch` may not be desirable for
 where various files on the volume may get random owners.
 
 With a single `VolumeChangePolicy`, user has to fall back to `Recursive` policy and SELinux labels would be unnecessarily changed.
+
+### Implement kubelet admission
+
+It's not really an alternative, but it's worth mentioning that we originally proposed to implement kubelet admission for Pods with SELinux contexts.
+
+Kubelet _could_ reject a Pod that uses a volume with a different SELinux context than the volume is currently mounted with on the node, right during Pod admission.
+Such a Pod would end in `Failed` state with appropriate message, and it would never reach kubelet internal states (e.g. DSW, ASW, MountDevice, SetUp, etc.).
+
+We thought it would send a clear message to the user that such a pod can't start, and why.
+During implementation and testing it has shown that it's very noisy. Consider this experiment:
+
+1. User runs a single-replica Deployment with no SELinux context set. The Deployment uses a single volume. The Deployment runs a Pod A on Node 1.
+1. User edits the Deployment and sets the SELinux context of the Pod, because they want to benefit from `SELinuxMount` feature.
+1. Deployment / ReplicaSet controllers start a new Pod B with the new SELinux context.
+1. Scheduler (accidentally?) puts Pod B to Node 1, where Pod A still runs.
+1. Kubelet rejects the new Pod, because the volume is already mounted without any SELinux context, while the new Pod wants a specific one.
+1. Deployment / ReplicaSet controllers start another Pod with the new SELinux context. Scheduler puts it to Node 1 again.
+1. Kubelet rejects the Pod again. This loop can continue forever, as longs as the scheduler picks Node 1 and Pod A still runs there.
+
+I got hundreds Pods just created and rejected in a minute or so. While it sends a clear message to the user that something is wrong, it's just too noisy for the API server.
+
+**As result, we decided to remove pod admission in kubelet from this KEP.**
+
+Expected behavior without Pod admission is that kubelet admits the Pod B and eventually its volumes reach DSW.
+The Pod B will be `ContainerCreating` until Pod A terminates and the volume can be mounted with the right SELinux context.
+The same behavior was already implemented for `ReadWriteOncePod` AccessMode.
```

Comment:
I see the reasoning for not to add the pod admission in kubelet and that makes sense.  Can you explain why it was proposed in the KEP earlier?  What are the benefits of keeping Pod A with the original context and rejecting Pod B with a different SELinux context?

Reply:
We thought it's how kubelet should work - it should reject Pods that cannot run on the node as soon as possible. Which is exactly the case of two pods using the same volume with two different SELinux contexts - only one of them can run.

In practice, it produces this weird endless loop.